parser_library/src/config/assembler_options.cpp

```diff
@@ -100,55 +223,43 @@ template<typename T>
 void optional_from_json(const nlohmann::json& j, std::string_view name, std::optional<T>& o)
 {
     if (auto it = j.find(name); it != j.end())
+    {
+        if (o.has_value())
+            throw nlohmann::json::other_error::create(
+                501, std::string("Parameter ").append(name).append(" redefined"), j);
+
         o = it->get<T>();
-    else
-        o.reset();
+    }
 }
 
 void to_json(nlohmann::json& j, const assembler_options& p)
 {
     j = nlohmann::json::object();
     optional_to_json(j, "GOFF", p.goff); // GOFF and XOBJECTS are synonyms
+    optional_to_json(j, "MACHINE", p.machine);
     optional_to_json(j, "OPTABLE", p.optable);
     optional_to_json(j, "PROFILE", p.profile);
     optional_to_json(j, "SYSPARM", p.sysparm);
     optional_to_json(j, "SYSTEM_ID", p.system_id);
 }
 
-namespace {
-std::optional<bool> get_goff_from_json(const nlohmann::json& j)
-{
-    bool present = false;
-    bool goff = false;
-    bool xobject = false;
-
-    if (auto it = j.find("GOFF"); it != j.end())
-    {
-        it->get_to(goff);
-        present = true;
-    }
-
-    if (auto it = j.find("XOBJECT"); it != j.end())
-    {
-        it->get_to(xobject);
-        present = true;
-    }
-    if (present)
-        return goff || xobject; // GOFF and XOBJECTS are synonyms
-    else
-        return std::nullopt;
-}
-} // namespace
-
 void from_json(const nlohmann::json& j, assembler_options& p)
 {
     if (!j.is_object())
-        throw nlohmann::json::other_error::create(501, "asm_options must be an object.");
+        throw nlohmann::json::other_error::create(501, "asm_options must be an object.", j);
 
-    p.goff = get_goff_from_json(j);
+    optional_from_json(j, "GOFF", p.goff);
+    optional_from_json(j, "MACHINE", p.machine);
     optional_from_json(j, "OPTABLE", p.optable);
     optional_from_json(j, "PROFILE", p.profile);
     optional_from_json(j, "SYSPARM", p.sysparm);
     optional_from_json(j, "SYSTEM_ID", p.system_id);
+    optional_from_json(j, "XOBJECT", p.goff);
+
+    // Machine and OPTABLE are basically synonyms
+    if (p.machine.has_value() && p.optable.has_value())
```

Comment:
assembler_options::valid() should catch this situation, right?

Reply:
It does. Nevertheless, it depends on what we want to tell the user in VSCode. The current code in PR results in the following when both options are used:

![image](https://user-images.githubusercontent.com/99467904/167101999-adba02b2-56a5-480c-b3d2-fc8bde47ee9c.png)


Without the check it results in this:
![image](https://user-images.githubusercontent.com/99467904/167102054-5ef48b8b-2379-4d4f-90c9-cebb401ddd1b.png)


Personally, I prefer the first diag message as the option isn't exactly invalid. What do you think?

Reply:
I think the second one is more accurate. Additionally, it mentions the name of the processor group, that the problem is related to assembler options.